benchmark/inclusion_compositional_field/inclusion_compositional_field.cc

```diff
@@ -0,0 +1,564 @@
+#include <aspect/material_model/simple.h>
+#include <aspect/velocity_boundary_conditions/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/global.h>
+
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/function_lib.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/vector_tools.h>
+
+
+
+namespace aspect
+{
+  /**
+   * This is the "Pure shear/Inclusion" benchmark defined in the following paper:
+   * @code
+   *  @Article{DMGT11,
+   *    author =       {T. Duretz and D. A. May and T. V. Gerya and P. J. Tackley},
+   *    title =        {Discretization errors and free surface stabilization in the
+   *                  finite difference and marker-in-cell method for applied
+   *                  geodynamics: {A} numerical study},
+   *    journal =      {Geochemistry Geophysics Geosystems},
+   *    year =         2011,
+   *    volume =       12,
+   *    pages =        {Q07004/1--26}}
+   * @endcode
+   *
+   * The results are published in Kronbichler, Heister and Bangerth paper.
```

Comment:
For all of the benchmarks involved in this PR, whether we are using compositional fields or the direct method (finite elements), the `Nonlinear solver scheme` used is `Stokes only`. Therefore, the material model for these benchmarks does not depend on temperature.  In the original `inclusion`benchmark for example, viscosity is explicitly defined as a function of position as part of the material model. Similarly, in the `inclusion_field` benchmark, viscosity is set to be the value of the compositional field at a given position as part of the material model as well. However, initially, the compositional field is intialized as a function of position.